torch/csrc/jit/tensorexpr/bounds_overlap.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <torch/csrc/jit/tensorexpr/ir_simplifier.h>
+#include <torch/csrc/jit/tensorexpr/ir_visitor.h>
+#include <torch/csrc/jit/tensorexpr/stmt.h>
+#include <deque>
+#include <vector>
+
+namespace torch {
+namespace jit {
+namespace tensorexpr {
+namespace bounds {
```

Comment:
would a reasonable compromise be to put this in the mem_dependency namespace?

Reply:
Sure, one namespace for these would definitely be better. How about an even more general one, say `analysis`, so that we can share the same namespace with all analysis?

Reply:
Sure I guess that works.